diode-server/reconciler/ingestion_processor_internal_test.go

```diff
@@ -0,0 +1,454 @@
+package reconciler
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"testing"
+
+	"github.com/netboxlabs/diode/diode-server/gen/diode/v1/diodepb"
+	"github.com/netboxlabs/diode/diode-server/netbox"
+	"github.com/netboxlabs/diode/diode-server/netboxdiodeplugin"
+	"github.com/netboxlabs/diode/diode-server/reconciler/changeset"
+	"github.com/redis/go-redis/v9"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+type MockRedisClient struct {
+	mock.Mock
+}
+type MockNbClient struct {
+	mock.Mock
+}
+
+func (m *MockNbClient) ApplyChangeSet(ctx context.Context, req netboxdiodeplugin.ChangeSetRequest) (*netboxdiodeplugin.ChangeSetResponse, error) {
+	args := m.Called(ctx, req)
+	return args.Get(0).(*netboxdiodeplugin.ChangeSetResponse), args.Error(1)
+}
+
+func (m *MockNbClient) RetrieveObjectState(ctx context.Context, params netboxdiodeplugin.RetrieveObjectStateQueryParams) (*netboxdiodeplugin.ObjectState, error) {
+	args := m.Called(ctx, params)
+	return args.Get(0).(*netboxdiodeplugin.ObjectState), args.Error(1)
+}
+func (m *MockRedisClient) Ping(ctx context.Context) *redis.StatusCmd {
+	args := m.Called(ctx)
+	return args.Get(0).(*redis.StatusCmd)
+}
+
+func (m *MockRedisClient) Close() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+func (m *MockRedisClient) XGroupCreateMkStream(ctx context.Context, stream, group, start string) *redis.StatusCmd {
+	args := m.Called(ctx, stream, group, start)
+	return args.Get(0).(*redis.StatusCmd)
+}
+
+func (m *MockRedisClient) XReadGroup(ctx context.Context, a *redis.XReadGroupArgs) *redis.XStreamSliceCmd {
+	args := m.Called(ctx, a)
+	return args.Get(0).(*redis.XStreamSliceCmd)
+}
+
+func (m *MockRedisClient) XAck(ctx context.Context, stream, group string, ids ...string) *redis.IntCmd {
+	args := m.Called(ctx, stream, group, ids)
+	return args.Get(0).(*redis.IntCmd)
+}
+
+func (m *MockRedisClient) XDel(ctx context.Context, stream string, ids ...string) *redis.IntCmd {
+	args := m.Called(ctx, stream, ids)
+	return args.Get(0).(*redis.IntCmd)
+}
+
+func (m *MockRedisClient) Do(ctx context.Context, args ...interface{}) *redis.Cmd {
+	callArgs := m.Called(ctx, args)
+	return callArgs.Get(0).(*redis.Cmd)
+}
```

Comment:
For `netboxdiodeplugin.NetBoxAPI` we already have generated in https://github.com/netboxlabs/diode/blob/develop/diode-server/netboxdiodeplugin/mocks/netboxapi.go so not needed here.

In general, these mocks should be generated for using [Mockery](https://vektra.github.io/mockery/latest/), apologies it's not documented (yet)

- installation
```shell
go install github.com/vektra/mockery/v2@v2.43.2
```

- update https://github.com/netboxlabs/diode/blob/develop/diode-server/.mockery.yaml by appending following to `packages`:
```
  github.com/netboxlabs/diode/diode-server/reconciler:
    config:
      all: true
      dir: "{{.InterfaceDir}}/mocks"
      outpkg: mocks
```
- generate mocks:
```shell
mockery
```

- update tests to import and use mocks

Reply:
Oh that is really cool. Just updated the PR using it! Great finding